gson/src/main/java/com/google/gson/JsonArray.java

```diff
@@ -30,367 +27,482 @@
+		if (o instanceof JsonElement)
+			return elements.remove(o);
+		return false;
+	}
+	
+	/**
+	 * Removes the element at the specified position in this array. Shifts any subsequent elements
+	 * to the left (subtracts one from their indices). Returns the element that was removed from
+	 * the array.
+	 *
+	 * @param index index the index of the element to be removed
+	 * @return the element previously at the specified position
+	 * @throws IndexOutOfBoundsException if the specified index is outside the array bounds
+	 * @since 2.3
+	 */
+	@Override
+	public JsonElement remove(int index) {
+		return elements.remove(index);
+	}
+	
+	@Override
+	public boolean removeAll(Collection<?> c) {
+		return elements.removeAll(c);
+	}
+	
+	@Override
+	public boolean retainAll(Collection<?> c) {
+		return elements.retainAll(c);
+	}
+	
+	/**
+	 * Returns the ith element of the array.
+	 *
+	 * @param i the index of the element that is being sought.
+	 * @return the element present at the ith index.
+	 * @throws IndexOutOfBoundsException if i is negative or greater than or equal to the
+	 * {@link #size()} of the array.
+	 */
+	@Override
+	public JsonElement get(int i) {
+		return elements.get(i);
+	}
+	
+	@Override
+	public void clear() {
+		elements.clear();
+	}
+	
+	@Override
+	public int indexOf(Object o) {
+		if (o instanceof JsonElement)
+			return elements.indexOf(o);
+		return -1;
+	}
+	
+	@Override
+	public int lastIndexOf(Object o) {
+		if (o instanceof JsonElement)
+			return elements.lastIndexOf(o);
+		return -1;
+	}
+
+	@Override
+	public boolean contains(Object o) {
+		if (o instanceof JsonElement)
+			return elements.contains(o);
+		return false;
+	}
+	
+	@Override
+	public boolean containsAll(Collection<?> c) {
+		return elements.containsAll(c);
+	}
+	
+	/** Returns the number of elements in this array. */
+	@Override
+	public int size() {
+		return elements.size();
+	}
+	
+	/** Returns true if this array is empty. */
+	@Override
+	public boolean isEmpty() { return elements.isEmpty(); }
+	
+	/**
+	 * Returns an iterator to navigate the elements of the array. Since the array is an ordered list,
+	 * the iterator navigates the elements in the order they were inserted.
+	 *
+	 * @return an iterator to navigate the elements of the array.
+	 */
+	@Override
+	public Iterator<JsonElement> iterator() {
+		return elements.iterator();
+	}
+	
+	@Override
+	public ListIterator<JsonElement> listIterator() {
+		return elements.listIterator();
+	}
+	
+	@Override
+	public ListIterator<JsonElement> listIterator(int index) {
+		return elements.listIterator(index);
+	}
+
+	@Override
+	public JsonElement[] toArray() {
+		return elements.toArray(new JsonElement[size()]);
+	}
+	
+	@Override
+	public <T> T[] toArray(T[] a) {
+		return elements.toArray(a);
+	}
+	
+	@Override
+	public List<JsonElement> subList(int fromIndex, int toIndex) {
+		return elements.subList(fromIndex, toIndex);
+	}
+	
+	@Override
+	public boolean equals(Object o) {
```

Comment:
Should get replaced by:
```
@Override
public boolean equals(Object o) {
	return o == this || o instanceof List<?> && ((List<?>) o).equals(elements);
}
```
but this is debatable because of backwards-compatibility.